src/babel/api/register/node.js

```diff
@@ -160,7 +161,7 @@ hookExtensions(util.canCompile.EXTENSIONS);
 
 export default function (opts = {}) {
   if (opts.only != null) only = util.arrayify(opts.only, util.regexify);
-  if (opts.ignore != null) ignore = util.arrayify(opts.ignore, util.regexify);
+  if (opts.ignore != null) ignore = util.arrayify(opts.ignore, util.regexify, isFunction);
```

Comment:
This approach also enriches the `arrayify` function. 
IMO, this is better since exclusions could come in handy in future